### PR TITLE
4.0.0-release-notes

### PR DESCRIPTION
**Pull request by hmottestad**
Javadocs, release-notes and news item for 4.0.0

### CODE REVIEW

File: site/content/release-notes/4.0.0.md [hunk 1]
```diff
@@ -2,10 +2,242 @@
...
+
+- Update to Java 11 as the minimally-required version of Java
+- Various improvements in the SparqlBuilder
+- Various speed and optimization improvements in the SPARQL query engine
+- Performance improvements in the SHACL engine
+- Performance improvements in the Memory Sail
+- Improved support for working with RDF4J in Spring and Spring Boot applications (see [documentation](/documentation/programming/spring) for details)
+- SeRQL and SPIN are no longer supported
+- Removal of several long-deprecated packages and interfaces
+- ...
```
Comment (jeenbroekstra) [inline site/content/release-notes/4.0.0.md:15]: I think we can remove this line.

File: site/content/release-notes/4.0.0.md [hunk 2]
```diff
@@ -2,10 +2,242 @@
...
+Removed support for defining the `writeStrategyFactory` through `FedXConfig`. Use `FedXFactory#withWriteStrategyFactory` instead.
+
+Removed deprecated support for specifying `fedxConfig` as a location in `FedXConfig`. Use `FedXConfig#setConfig` instead.
+
 ## Acknowledgements
 
-This release was made possible by contributions from ???
+This release was made possible by contributions from Andreas Schwarte, Florian
+Kleedorfer, Jerven Bolleman, Havard M. Ottestad, Bart Hanssens, Nils Hoffman,
+and Jeen Broekstra.
```
Comment (jeenbroekstra) [inline site/content/release-notes/4.0.0.md:243]: We should add Ken Wenzel, Antoine Willerval, and Nils Hoffman

File: site/content/news/rdf4j-400.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+title: "RDF4J 4.0.0 released"
+date: 2022-04-22T21:07:16+0200
+layout: "single"
+categories: ["news"]
+---
+
```
Comment (jeenbroekstra) [inline site/content/news/rdf4j-400.md:7]: Need a hand filling this out?

Comment (hmottestad) [reply site/content/news/rdf4j-400.md:7]: As always I appreciate your help with this :)

File: site/content/release-notes/4.0.0.md (hunk 1 above)
Comment (kenwenzel) [reply site/content/release-notes/4.0.0.md:15]: When should 4.0.0 be releases? It would be best if the thread bug for LMDB is fixed first.

Comment (hmottestad) [reply site/content/release-notes/4.0.0.md:15]: It's pushed to maven already. Only thing left is to make the announcement on our site.

Comment (hmottestad) [reply site/content/release-notes/4.0.0.md:15]: But we can push 4.0.1 within just a few days if we want.

File: site/content/news/rdf4j-400.md (hunk 3 above)
Comment (jeenbroekstra) [reply site/content/news/rdf4j-400.md:7]: Ok I'll write a blurb and push it.

Comment (jeenbroekstra) [reply site/content/news/rdf4j-400.md:9]: Done. I added some balloons (because, you know, we can).

**Comment by jeenbroekstra** [comment]
Pushed. I'll leave it to you to give it the final nod of approval and merge it.

**Comment by hmottestad** [comment]
> Pushed. I'll leave it to you to give it the final nod of approval and merge it.

Thanks so much Jeen!

**Pull request merged**